### PR TITLE
security workflow

### PR DESCRIPTION
**Pull request by hendrikebbers**
Provides security workflow in that client specifies keyckloak realm & app

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/887)

### CODE REVIEW

**Comment by hendrikebbers** [comment]
fix for https://github.com/canoo/dolphin-platform/issues/858

File: platform-examples/security-sample/security-client-javafx/src/main/java/com/canoo/platform/samples/security/Client.java [hunk 1]
```diff
@@ -17,42 +17,27 @@
...
+import com.canoo.platform.core.SimpleConfiguration;
 import com.canoo.platform.core.http.HttpClient;
 
+import static com.canoo.dp.impl.security.SecurityConfiguration.APPLICATION_PROPERTY_NAME;
+import static com.canoo.dp.impl.security.SecurityConfiguration.REALM_PROPERTY_NAME;
+
 public class Client {
 
-    public static void main(String[] args) throws Exception{
+    public static void main(String[] args) throws Exception {
```
Comment (giftkugel) [inline platform-examples/security-sample/security-client-javafx/src/main/java/com/canoo/platform/samples/security/Client.java]: Add `final` keyword.

File: platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/SimpleConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    public void setListProperty(final String key, final List<String> values) {
+        if (values == null) {
+            setProperty(key, null);
+        } else if (values.isEmpty()) {
+            setProperty(key, "");
+        } else {
+            final StringBuilder builder = new StringBuilder();
+            for (String value : values) {
```
Comment (giftkugel) [inline platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/SimpleConfiguration.java]: Add `final` keyword.

File: platform/dolphin-platform-server-javaee/src/main/java/com/canoo/dp/impl/server/javaee/DolphinPlatformJavaeeBootstrap.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
 
     @Override
     public void onStartup(final Set<Class<?>> c, final ServletContext servletContext) throws ServletException {
-        DefaultPlatformConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
+        ServerConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
```
Comment (giftkugel) [inline platform/dolphin-platform-server-javaee/src/main/java/com/canoo/dp/impl/server/javaee/DolphinPlatformJavaeeBootstrap.java]: Add `final` keyword.

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java [hunk 4]
```diff
@@ -51,11 +51,11 @@
     private Environment environment;
 
     @Autowired(required = false)
-    private DefaultPlatformConfiguration injectedConfig;
+    private ServerConfiguration injectedConfig;
 
     @Override
     public void onStartup(final ServletContext servletContext) throws ServletException {
-        DefaultPlatformConfiguration configuration = injectedConfig;
+        ServerConfiguration configuration = injectedConfig;
```
Comment (giftkugel) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: Add `final` keyword.

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java [hunk 5]
```diff
@@ -64,7 +64,7 @@ public void onStartup(final ServletContext servletContext) throws ServletExcepti
         bootstrap.init(servletContext, configuration);
     }
 
-    private void updateConfigurationBySpring(DefaultPlatformConfiguration configuration) {
+    private void updateConfigurationBySpring(ServerConfiguration configuration) {
```
Comment (giftkugel) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: Add `final` keyword.

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java [hunk 6]
```diff
@@ -64,7 +64,7 @@ public void onStartup(final ServletContext servletContext) throws ServletExcepti
...
         for(String key : configuration.getPropertyKeys()) {
```
Comment (giftkugel) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: Add `final` keyword.

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java [hunk 7]
```diff
@@ -64,7 +64,7 @@ public void onStartup(final ServletContext servletContext) throws ServletExcepti
...
             String valInSpringConfig = environment.getProperty(PREFIX + key);
```
Comment (giftkugel) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java [hunk 8]
```diff
@@ -31,7 +31,7 @@
     public void testConfigLoad() {
         try {
             //given:
-            DefaultPlatformConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
+            ServerConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java [hunk 9]
```diff
@@ -45,7 +45,7 @@ public void testConfigLoad() {
     public void testConfigurationProvider() {
         try {
             //given:
-            DefaultPlatformConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
+            ServerConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java [hunk 10]
```diff
@@ -61,7 +61,7 @@ public void testConfigurationProvider() {
     public void testConfigurationProviderDoNotOverwrite() {
         try {
             //given:
-            DefaultPlatformConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
+            ServerConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java [hunk 11]
```diff
@@ -75,7 +75,7 @@ public void testConfigurationProviderDoNotOverwrite() {
     public void testNullPropertyWillNotBeAdded() {
         try {
             //given:
-            DefaultPlatformConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
+            ServerConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java [hunk 12]
```diff
@@ -95,7 +95,7 @@ public void testNullPropertyWillNotBeAdded() {
     public void testNullPropertyWillNotRemoveOldValue() {
         try {
             //given:
-            DefaultPlatformConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
+            ServerConfiguration configuration = ConfigurationFileLoader.loadConfiguration();
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java [hunk 13]
```diff
@@ -27,13 +27,13 @@
 
     @Test
     public void testCommaSeparatedStringWithValidList(){
-        CrossSiteOriginFilter crossSiteOriginFilter = new CrossSiteOriginFilter(new DefaultPlatformConfiguration());
+        CrossSiteOriginFilter crossSiteOriginFilter = new CrossSiteOriginFilter(new ServerConfiguration());
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java [hunk 14]
```diff
@@ -27,13 +27,13 @@
...
         String commaSeparatedList = crossSiteOriginFilter.getAsCommaSeparatedList(Arrays.asList("origin", "authorization", "accept"));
         assertEquals("origin,authorization,accept",commaSeparatedList);
     }
     @Test(expectedExceptions = NullPointerException.class)
     public void testCommaSeparatedList(){
-        CrossSiteOriginFilter crossSiteOriginFilter = new CrossSiteOriginFilter(new DefaultPlatformConfiguration());
+        CrossSiteOriginFilter crossSiteOriginFilter = new CrossSiteOriginFilter(new ServerConfiguration());
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/DolphinDefaultPlatformConfigurationTest.java [hunk 15]
```diff
@@ -26,7 +26,7 @@
     @Test
     public void testDefaultConfiguration() {
         //given:
-        DefaultPlatformConfiguration configuration = new DefaultPlatformConfiguration();
+        ServerConfiguration configuration = new ServerConfiguration();
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/DolphinDefaultPlatformConfigurationTest.java]: Add `final` keyword.

File: platform-examples/security-sample/security-client-javafx/src/main/java/com/canoo/platform/samples/security/Client.java (hunk 1 above)
Comment (hendrikebbers) [reply platform-examples/security-sample/security-client-javafx/src/main/java/com/canoo/platform/samples/security/Client.java]: done

File: platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/SimpleConfiguration.java (hunk 2 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-core/src/main/java/com/canoo/platform/core/SimpleConfiguration.java]: done

File: platform/dolphin-platform-server-javaee/src/main/java/com/canoo/dp/impl/server/javaee/DolphinPlatformJavaeeBootstrap.java (hunk 3 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server-javaee/src/main/java/com/canoo/dp/impl/server/javaee/DolphinPlatformJavaeeBootstrap.java]: done

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java (hunk 4 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: done

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java (hunk 5 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: done

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java (hunk 6 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: done

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java (hunk 7 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server-spring/src/main/java/com/canoo/dp/impl/server/spring/DolphinPlatformSpringBootstrap.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java (hunk 8 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java (hunk 9 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java (hunk 10 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java (hunk 11 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java (hunk 12 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/ConfigurationFileLoaderTest.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java (hunk 13 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java (hunk 14 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/CrossSiteOriginFilterTest.java]: done

File: platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/DolphinDefaultPlatformConfigurationTest.java (hunk 15 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/test/java/com/canoo/impl/server/DolphinDefaultPlatformConfigurationTest.java]: done

**Pull request merged**